core/src/test/java/org/elasticsearch/index/shard/IndexShardIT.java

```diff
@@ -506,6 +517,65 @@ public void postDelete(ShardId shardId, Engine.Delete delete, Engine.DeleteResul
         }
     }
 
+    /** Check that the accounting breaker correctly matches the segments API for memory usage */
```

Comment:
any chance we can get this into `IndexShardTests` as a real unittest?

Reply:
Sure, I'll work on a unittest for this

Reply:
Added in 8308df014b7aebd33404eab6595153349ed1a160